eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/io/DbsnpJsonClusteredVariantsWriter.java

```diff
@@ -51,16 +49,18 @@ public DbsnpJsonClusteredVariantsWriter(MongoTemplate mongoTemplate, ImportCount
     @Override
     public void write(List<? extends DbsnpClusteredVariantEntity> clusteredVariants) {
         try {
-            dbsnpClusteredVariantWriter.write(clusteredVariants);
+            if (!clusteredVariants.isEmpty()) {
+                dbsnpClusteredVariantWriter.write(clusteredVariants);
+            }
         } catch (BulkOperationException exception) {
-            List<String> ids = extractUniqueHashesForDuplicateKeyError(exception).collect(toList());
-            Query query = new Query();
-            query.addCriteria(Criteria.where("_id").in(ids));
-            List<DbsnpClusteredVariantEntity> entities = mongoTemplate.find(query, DbsnpClusteredVariantEntity.class);
-            List<Long> rsIDs = entities.stream()
-                .map(DbsnpClusteredVariantEntity::getAccession)
-                .collect(toList());
-            logger.error("Duplicate RS IDs: {}", rsIDs);
+            List<String> hashes = extractUniqueHashesForDuplicateKeyError(exception).collect(toList());
+            List<String> variantsThatFailedInsert =
+                    clusteredVariants
```

Comment:
Given that the code is using the list of `clusteredVariants` instead of querying mongo (which I think is better) please delete the `MongoTemplate` object and its dependency

Reply:
But the constructor for DbsnpCVWriter needs the MongoTemplate...

Reply:
But the `MongoTemplate` is not really used here, just for creating the `DbsnpClusteredVariantWriter`. I think you should inject a `DbsnpClusteredVariantWriter` (which is what you really need) instead of a `MongoTemplate`